platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerImpl.java

```diff
@@ -93,36 +60,36 @@ public FileSystemManagerImpl(
             rethrowIO(() -> Files.createDirectories(rootPath));
         }
 
-        this.tempPath = rootPath.resolve(tmpDirName);
-        this.savedPath = rootPath.resolve(dataDirName);
+        this.dataPath = rootPath.resolve(dataDirName);
+        this.tmpPath = rootPath.resolve(tmpDirName);
 
-        if (!exists(savedPath)) {
-            rethrowIO(() -> Files.createDirectory(savedPath));
+        if (!exists(dataPath)) {
+            rethrowIO(() -> Files.createDirectory(dataPath));
         }
-        if (exists(tempPath)) {
-            rethrowIO(() -> FileUtils.deleteDirectory(tempPath));
+        if (exists(tmpPath)) {
+            rethrowIO(() -> FileUtils.deleteDirectory(tmpPath));
         }
-        rethrowIO(() -> Files.createDirectory(tempPath));
+        rethrowIO(() -> Files.createDirectory(tmpPath));
     }
 
     /**
      * Resolve a path relative to the{@code savedPath} of this file system manager.
      *
      * @param relativePath the path to resolve against the root directory
      * @return the resolved path
-     * @throws IllegalArgumentException if the path is "above" the root directory (e.g. resolve("../foo")
+     * @throws IllegalArgumentException if the path is "above" the root directory (e.g. resolve("../foo"))
      */
     @NonNull
     @Override
     public Path resolve(@NonNull final Path relativePath) {
-        return requireValidSubPathOf(savedPath, savedPath.resolve(relativePath));
+        return requireValidSubPathOf(dataPath, dataPath.resolve(relativePath));
     }
 
     /**
      * Creates a path relative to the {@code tempPath} directory of the file system manager.
      * There is no file or directory actually being created after the invocation of this method.
-     * All calls to this method will return a different path even if {@code tag} is not set.
-     * A separate instance pointing to the same {@code rootPath} can create the same paths and should be managed outside this class.
+     * A call to this method will return a different path even if {@code tag} is not set for the same instance of this class.
+     * Two instances of this class pointing to the same root directory can generate repeated or existing paths as result of this method invocation.
```

Comment:
This is actually a problem. When I tried to change MerkleDb to use `FileSystemManager`, some tests failed with really weird exceptions, when file system manager couldn't delete a dir on init, because the dir was not empty. Debugging was nightmare, but in the end I found the problem was there were multiple instances of `FileSystemManager` with the same root/data/tmp dir. This is exactly what the comment above describes.

What if we store file system manager state (e.g. tmp file name index) per root, not per manager, with proper synchronization? It would let multiple file system manager instances with the same root dir co-exist nicely. @mxtartaglia-sl What do you think about it?